Modules/Scripted/DICOM/DICOM.py

```diff
@@ -329,7 +343,24 @@ def setup(self):
     self.dicomFrame.setText("DICOM Database and Networking")
     self.layout.addWidget(self.dicomFrame)
 
-    self.detailsPopup = self.getSavedDICOMDetailsWidgetType()()
+    # Create widget and add a new DICOM browser layout
+    self.viewFactory = slicer.qSlicerSingletonViewFactory()
+    self.viewFactory.setTagName("dicombrowser")
+
+    self.detailsPopup = None
+    self.setDetailsPopup(self.getSavedDICOMDetailsWidgetType()())
+
+    layoutManager = slicer.app.layoutManager()
+    layoutManager.registerViewFactory(self.viewFactory)
```

Comment:
The layout should be registered right at application startup, because if the scene is saved with DICOM layout active then when we open the scene it would try to switch to that layout.

By the way - does the browser work correctly if someone just activates the DICOM layout but does not switch to the DICOM module?

Reply:
Yes, calling set layout and calling a layout that has the dicombrowser tag displays a functional DICOM browser widget.

Reply:
Isn't it a problem that autofillbackground is not set in this case?